.github/gha_monitor/gha_failures.py

```diff
@@ -0,0 +1,121 @@
+ONCALL_WIKI = "https://w.amazon.com/bin/view/AWSCryptography/AWSCryptoTools/TransportLibraries/OnCall/"
+
+
+class GitHub_Actions(github.GitHub_Client):
+    params = {
+        # Needed when using an API key
+        'github_username': os.getenv('github_username'),
+        'github_password': os.getenv('github_password'),
+        # Use from within an Action - ignored if username set
+        'token': os.getenv('GITHUB_TOKEN'),
+        'repo_organization': os.getenv('GITHUB_REPO_ORG'),
+        'repo': os.getenv('GITHUB_REPO')
+    }
+
+
+class S2n_Notices(sns.SNS_Client):
+    params = {
+        'topic_arn': 'arn:aws:sns:us-west-2:024603541914:s2n_notices'
+    }
+
+
+def workflow_filter(log):
+    """ Reduce the number of elements from GitHub for JSON SNS. """
+    top_level_keys = set(['head_branch', 'head_commit', 'status', 'conclusion', 'html_url', 'created_at', 'repository'])
+    to_drop = set(log) - top_level_keys
+    for element in to_drop:
+        del log[element]
+    logging.debug('Filtered json: {log}')
+    return log
+
+
+def message_text():
+    """ Formatting for text message. """
+    return """
+s2n GitHub Action monitor notice
+State: failed
+Repo: {repo}
+GHA failure time: {time}
+Workflow name: {workflow_name}
+Wiki: {wiki}
+URL: {url}
+started by: {commit_owner}\n
+"""
+
+
+def main():
+    """ Main entrypoint. """
+    logging.info('Starting up')
+    plaintext_notice = []
+    json_notice = []
+    gh_api = GitHub_Actions()
+    s2n_text_client = S2n_Notices()
+    #s2n_json_client = S2n_Json_Notices()
+
+    # Get the Action workflow log from the Github API
+    gh_api.get_workflow_log_chunk(final_state='failure')
+    logging.info(f"Looking for events newer than {TIME_WINDOW_BEGIN}")
+    if gh_api.worklog:
+        for i in gh_api.worklog:
+            # We're going to lookup some of the API URLs
+            enhanced_worklog = i
+
+            # Parse the event date/time so we can compare it
+            datetime_creation = parser.parse(i['created_at'])
+            logging.debug(f"looking at event from {enhanced_worklog['created_at']}")
+
+            # If the event is recent enough, process it.
+            if datetime_creation > TIME_WINDOW_BEGIN:
+                logging.debug("Workflow_url: " + enhanced_worklog['workflow_url'])
+                # The name of the workflow isn't in the failure object, look it up.
+                enhanced_worklog['workflow_name'] = gh_api.get_workflow_name(i['workflow_url'].split('/')[-1:][0])
+                enhanced_worklog['repo'] = gh_api.params['repo']
+
+                # Construct a notification string.
+                notice_msg = message_text().format(
+                    time=enhanced_worklog['created_at'],
+                    url=enhanced_worklog['html_url'],
+                    commit_owner=enhanced_worklog['head_commit']['author']['email'],
+                    repo=enhanced_worklog['repo'],
+                    wiki=ONCALL_WIKI,
```

Comment:
We don't really require this link here, lets get rid of it.

Reply:
removed